realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +283,40 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} is called from other threads.
+     *
+     * @return {@code true} if transactions are committed to this Realm instance, {@code false} if
+     * wait has been cancelled by {@link #stopWaitForChange()} called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (handlerController != null) {
+            throw new IllegalStateException("Cannot wait for changes inside of a Looper.");
+        }
+        sharedGroupManager.setWaitForChangeEnabled(true);
+        boolean hasChanged = sharedGroupManager.waitForChange();
+        if (hasChanged) {
+            sharedGroupManager.advanceRead();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes {@link #waitForChange()} return {@code false} immediately. This method is threadsafe
+     * and should be called from another thread than the one that called {@link #waitForChange}.
+     * If {@link #waitForChange()} is called on a Realm in another thread after this method is called,
```

Comment:
Yes, `waitForChange()` is indeed a very powerful and dangerous method. We should write a blog post explaining when it can go wrong.